src/ld65/config.c

```diff
@@ -2107,10 +2107,12 @@ unsigned CfgProcess (void)
             Addr += S->Seg->Size;
 
             /* If this segment will go out to the file, or its place
-            ** in the file will be filled, then increase the file size.
+            ** in the file will be filled, then increase the file size,
+            ** unless it's an OVERWRITE segment.
             */
             if (S->Load == M &&
-                ((S->Flags & SF_BSS) == 0 || (M->Flags & MF_FILL) != 0)) {
+                ((S->Flags & SF_BSS) == 0 || (M->Flags & MF_FILL) != 0)
+                    && (S->Flags & SF_OVERWRITE) == 0) {
```

Comment:
Please format the multi-line expression like e.g. here: https://github.com/cc65/cc65/blob/master/src/ld65/config.c#L2013-L2014